src/lib/y2storage/filesystems/legacy_nfs.rb

```diff
@@ -0,0 +1,230 @@
+        graph = check_devicegraph(devicegraph)
+
+        nfs = find_nfs_device(graph)
+        nfs.mountpoint = mountpoint
+        nfs.fstab_options = fstopt.split(/[\s,]+/)
+      end
+
+      # Finds the equivalent {Nfs} object in the devicegraph
+      #
+      # It first tries to match the object by #old_server and #old_path and, if
+      # that fails, by #server and #path.
+      #
+      # @raise [ArgumentError] if no devicegraph is given and no default
+      #   devicegraph has been previously defined
+      #
+      # @param devicegraph [Devicegraph, nil] if nil, the default devicegraph
+      #   will be used
+      # @return [Nfs, nil] found device or nil if no device matches
+      def find_nfs_device(devicegraph = nil)
+        graph = check_devicegraph(devicegraph)
+
+        old = nil
+        if share_changed?
+          old = Nfs.find_by_server_and_path(graph, old_server, old_path)
+        end
+        old || Nfs.find_by_server_and_path(graph, server, path)
+      end
+
+      # Whether this represents an NFS mount in which the remote share (server
+      # and/or remote path) has changed
+      #
+      # In fact, changing the connection information implies replacing the
+      # NFS mount with a new different one.
+      #
+      # @see #old_server
+      # @see #old_path
+      #
+      # @return [Boolean]
+      def share_changed?
+        !old_server.nil?
+      end
+
+      # @return [String]
+      def inspect
+        "<LegacyNfs attributes=#{to_hash}>"
+      end
+
+      # @see .new_from_hash
+      def initialize_from_hash(attributes)
+        @server, @path = split_share(attributes["device"])
+        @mountpoint = attributes["mount"]
+        @fstopt = attributes["fstopt"]
+
+        old_share = attributes["old_device"]
+        return if old_share.nil? || old_share.empty?
+        @old_server, @old_path = split_share(old_share)
+      end
+
+      # @see .new_from_nfs
+      def initialize_from_nfs(nfs)
+        @server     = nfs.server
+        @path       = nfs.path
+        @mountpoint = nfs.mountpoint
+        @fstopt     = nfs.fstab_options.empty? ? "defaults" : nfs.fstab_options.join(",")
+      end
+
+      # String representing the remote NFS share, as specified in fstab
+      #
+      # @return [String]
+      def share
+        share_string(server, path)
+      end
+
+    protected
+
+      # Breaks a string representing a share, in the format used in fstab, into
+      # its two components (server and path)
+      def split_share(share_string)
+        share_string.split(":")
+      end
+
+      # Composes a string to represent a share, in the format used in fstab,
+      # from its two components
+      def share_string(svr, dir)
+        "#{svr}:#{dir}"
+      end
+
+      # Ensures there is a devicegraph to work with, raising an exception
+      # otherwise
+      def check_devicegraph(devicegraph)
+        result = devicegraph || default_devicegraph
+        if result.nil?
```

Comment:
np: You could just do `return result if result` and avoid nesting. Just a matter of preference.